scope.go

```diff
@@ -387,16 +431,13 @@ func (s *scope) subscope(prefix string, immutableTags map[string]string) Scope {
 	immutableTags = mergeRightTags(s.tags, immutableTags)
 	key := scopeRegistryKey(prefix, immutableTags)
 
-	s.registry.RLock()
+	s.registry.Lock()
```

Comment:
we currently have `existing, ok := s.registry.subscopes[key]` twice below, which is probably a remnant of checking under rlock vs lock.

To avoid any performance difference, suggest doing the same thing as the counters etc by having the rlock check first in a separate function.

Reply:
done!